packages/root-password-protect/src/plugin/plugin.tsx

```diff
@@ -172,10 +192,14 @@ async function verifySessionCookie(
 export async function renderPasswordPage(
   req: Request,
   res: Response,
+  protectedRoute: PasswordProtectedRoute,
   props?: Omit<PasswordPageProps, 'nonce'>
 ) {
+  const [customRoute] =
+    req.renderer?.getRoute(protectedRoute.unauthorizedPageUrl || '/401') ?? [];
+  const Component = customRoute?.module.default ?? PasswordPage;
   const nonce = generateNonce();
-  const mainHtml = renderToString(<PasswordPage {...props} nonce={nonce} />);
+  const mainHtml = renderToString(<Component {...props} nonce={nonce} />);
```

Comment:
when using `req.renderer.getRoute()`, you'll need to use the renderer to render the html, otherwise if there are custom components (and other root-specific rendering logic) it wouldn't get picked up.

the renderToString() method here should only be used with the PasswordPage which renders the html directly using normal preact rendering (without the root magic).